### DISCUSSION TITLE
How to extend the ceph filesystem if we reach the maximum size of OSD disks ?

### DISCUSSION THREAD

**Discussion by ahmedTouati** [rook/rook#7367] [General]
I'm running a rook-ceph cluster with 3 OSDs which span 3 EBSs, everything is running as expected but we know that the maximum size of those EBSs is 16TBs, we presume that in the future we'll hit that limit, in this case, all OSDs will be filled up, we think of using more OSDs  with grouped replication but i don't know if that possible.

**Comment by travisn** [comment]
@ahmedTouati How did you create the OSDs? When running in AWS, it's recommended to use the `storageClassDeviceSets` as found in the cluster-on-pvc.yaml example. This way you can add more OSDs simply by increasing the `count` of the device set, instead of expanding the individual OSD size. 
For future questions, please see the [Discussions](https://github.com/rook/rook/discussions), it is preferred instead of opening an issue, thanks.

**Comment by ahmedTouati** [comment]
> @ahmedTouati How did you create the OSDs? When running in AWS, it's recommended to use the `storageClassDeviceSets` as found in the cluster-on-pvc.yaml example. This way you can add more OSDs simply by increasing the `count` of the device set, instead of expanding the individual OSD size.
> For future questions, please see the [Discussions](https://github.com/rook/rook/discussions), it is preferred instead of opening an issue, thanks.

@travisn we are using the **cluster-on-pvc.yaml** with **storageClassDeviceSets** already but what is the difference if we add additional OSDs since all data is replicated across them or you mean using more OSDs than the actual **Replicated Data Pool** ??

**Reply by travisn** [reply]
The number of OSDs in the cluster is independent from the pool `replicated.size` setting. If you add more OSDs, you'll get more capacity, and you won't be limited by the EBS volume size.

**Reply by ahmedTouati** [reply]
@travisn  what if i wanted to control OSD reservation for Data pools, so we can use multiple filesystems each with different data pool like the picture below
[http://karan-mj.blogspot.fi/2014/01/how-data-is-stored-in-ceph-cluster.html]()

**Reply by travisn** [reply]
@ahmedTouati Why do you need to control the OSD corresponding to data pools? You want to have greater isolation between filesystems? 

This could be done by with Ceph's deviceClass. For example, set the [crushDeviceClass](https://github.com/rook/rook/blob/release-1.5/cluster/examples/kubernetes/ceph/cluster-on-pvc.yaml#L122) on a storageClassDeviceSet to be some unique name, then when you create the filesystem, set the `deviceClass` on the data pool to the same name as the OSDs created in one of the storageClassDeviceSets.

**Reply by travisn** [reply]
But unless you really need isolation, you shouldn't care which OSDs are used by the pools.

**Reply by ahmedTouati** [reply]
@travisn OK thanks a lot
